lib/cli/run.js

```diff
@@ -151,6 +152,14 @@ exports.builder = yargs =>
         description: 'Inverts --grep and --fgrep matches',
         group: GROUPS.FILTERS
       },
+      jobs: {
+        description:
+          'Number of concurrent jobs for --parallel; use 1 to run in serial',
+        defaultDescription: '(number of CPU cores - 1)',
+        requiresArg: true,
+        group: GROUPS.RULES,
+        default: Math.max(2, coreCount - 1)
```

Comment:
hmmm.  it should be documented, at minimum (https://github.com/mochajs/mocha/pull/4245/files#diff-0ae2ed2e6fe54e5add4d6de80324e23bR158)

I think similar logic is duplicated elsewhere anyway.  if it is, I think L161 can safely be removed.

it's true that we can't just pick a static number that's going to work for all cases.  however, that doesn't mean we shouldn't provide reasonable defaults, even if those defaults are situational.

I suppose I'm not clear on what you'd prefer the alternative to be?

Reply:
upon further inspection, if we _don't_ set the value at least before instantiating our `Mocha` instance for the first time, we'll need to explicitly enable parallel mode, because `jobs` must be greater than 1.  If we keep a default like this, then we can let the constructor decide whether to enable it, based on the options.

Reply:
Yes it should be documented. However, it is fine to keep the default `undefined` and decide that exact value later. Magic defaults don't translate to something like JSON schema (#3727) if we decide to go that route (I think we should).  

Can't we simply set the value to `Math.max(2, coreCount - 1)` if the value is `undefined` in the mocha constructor? (or mixing the default here) https://github.com/mochajs/mocha/blob/b273286bed8b6e4cebceeed5ce87cabf2f05a3b9/lib/mocha.js#L129)

Reply:
looks like I missed this one

Reply:
OK, this should be more to your liking now